webrtc.html

```diff
@@ -3088,25 +3101,96 @@ <h2>Methods</h2>
+                            was created, then let <var>p</var> be the result of
+                            <a href="#dfn-create-an-offer">creating an offer</a>
+                            with <var>connection</var>, and return the result of
+                            <a href="https://w3ctag.github.io/promises-guide/#transforming-by">
+                            transforming</a> <var>p</var> with a fulfillment
+                            handler that returns the result of
+                            <a href="#set-description">setting the RTCSessionDescription</a>
+                            indicated by its first argument.</p>
+                          </li>
+                        </ol>
+                      </li>
+                      <li>
+                        <p>If <var>sdp</var> is the empty string, and
+                          <var>type</var> is <code>"answer"</code> or
+                          <code>"pranswer"</code>, then run the following sub
+                          steps:</p>
+                        <ol>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>Set <var>sdp</var> to the value of
+                            <var>connection</var>'s <a>[[\LastCreatedAnswer]]</a>
+                            slot.</p>
+                          </li>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>If <var>sdp</var> is the empty string, or if it
+                            no longer accurately represents the system state of
+                            <var>connection</var>, or if <var>connection</var>'s
+                            configured indentity provider has changed since
+                            <var>connection</var>'s <a>[[\LastCreatedAnswer]]</a>
+                            was created, then let <var>p</var> be the result of
+                            <a href="#dfn-create-an-answer">creating an answer</a>
+                            with <var>connection</var>, and return the result of
+                            <a href="https://w3ctag.github.io/promises-guide/#transforming-by">
+                            transforming</a> <var>p</var> with a fulfillment
+                            handler that runs the following steps:</p>
+                            <ol>
+                              <li>
+                                <p>Let <var>answer</var> be the handler's first
+                                argument.</p>
+                              </li>
+                              <li>
+                                <p>Return the result of <a
+                                href="#set-description">setting the RTCSessionDescription</a>
+                                indicated by
+                                <code>{<var>type</var>, <var>answer</var>.sdp}</code>.
+                                </p>
```

Comment:
Can we phrase the create-offer-and-SLD and create-answer-and-SLD with the same language? (Why are sub-steps needed here but not above?)

Reply:
It's different, because here we need to use the passed-in *type*, which in the case of pranswer is `"pranswer"`, ***not*** `"answer"` which is what the [**create an answer**](https://rawgit.com/jan-ivar/webrtc-pc/glareproof/webrtc.html#dfn-create-an-answer) algorithm returns.

Reply:
Ack